model/monitorCache/implementation/DeliveryMonitoringData.php

```diff
@@ -77,16 +78,23 @@ class DeliveryMonitoringData implements DeliveryMonitoringDataInterface, Service
     ];
 
     /**
-     * DeliveryMonitoringData constructor.
      * @param DeliveryExecutionInterface $deliveryExecution
+     * @param $data
+     * @throws \common_exception_NotFound
      */
     public function __construct(DeliveryExecutionInterface $deliveryExecution, $data)
     {
         $this->deliveryExecution = $deliveryExecution;
         if (is_array($data) && !empty($data)) {
             $this->data = $data;
         } else {
-            $this->data = [DeliveryMonitoringService::DELIVERY_EXECUTION_ID => $deliveryExecution->getIdentifier()];
+            $this->data = [
+                DeliveryMonitoringService::DELIVERY_EXECUTION_ID => $deliveryExecution->getIdentifier(),
+            ];
+        }
+
+        if (!array_key_exists(DeliveryMonitoringService::STATUS, $this->data) && $deliveryExecution->getState()) {
```

Comment:
Why we need set execution state in constructor?
Even if we need it I believe we can completely remove `if` condition.